packages/flutter/lib/src/widgets/debug.dart

```diff
@@ -249,6 +250,54 @@ bool debugItemsHaveDuplicateKeys(Iterable<Widget> items) {
   return false;
 }
 
+/// Asserts that the given context has a [SplashBox] ancestor within the closest
+/// [LookupBoundary].
+///
+/// Used by many widgets to make sure that they are only used in contexts where
+/// they have a [SplashBox] or other widget that enables [Splash] effects.
+///
+/// To call this function, use the following pattern, typically in the
+/// relevant Widget's build method:
+///
+/// ```dart
+/// assert(debugCheckSplash(context));
+/// ```
+///
+/// Always place this before any early returns, so that the invariant is checked
+/// in all cases. This prevents bugs from hiding until a particular codepath is
+/// hit.
+///
+/// This method can be expensive (it walks the element tree).
+///
+/// Does nothing if asserts are disabled. Always returns true.
+bool debugCheckSplash(BuildContext context) {
+  assert(() {
+    if (Splash.maybeOf(context) == null) {
+      final bool hiddenByBoundary = LookupBoundary.debugIsHidingAncestorRenderObjectOfType<SplashController>(context);
+      throw FlutterError.fromParts(<DiagnosticsNode>[
+        ErrorSummary('No SplashController found${hiddenByBoundary ? ' within the closest LookupBoundary' : ''}.'),
+        if (hiddenByBoundary)
+          ErrorDescription(
+            "There is an ancestor SplashController, but it's hidden by a LookupBoundary.",
+          ),
+        ErrorDescription(
+          '${context.widget.runtimeType} widgets use a SplashController to show '
+          'Splash effects, and no SplashController ancestor was found within '
+          'the closest LookupBoundary.\n',
+        ),
+        ErrorHint(
+          'A SplashController can be provided by an ancestor SplashBox; alternatively, '
+          'there are several viable options from the Material libary, including '
+          'Material, Card, Dialog, Drawer, and Scaffold.',
+        ),
```

Comment:
If someone is not using the material library, I am not sure if this hint is very helpful.

Reply:
Fair point, though my guess would be that the majority of the time, the person reading this message is someone using the material library.

Reply:
Then why are we moving this to the widgets library? :)

Reply:
Once this migration takes place, I imagine there will be Cupertino widgets that build a SplashBox; once that happens, I think it'd be good to add them to this message.

Reply:
Would we further alter the error message to account for every widget and library possible? The point I was trying to make was that error messages should be as specific as they can be. Listing widgets that may or may not have to do with the error is not helpful to the user.

Reply:
+1 on this being a little too specific. You can say something like, "Many existing widgets in Flutter already contain a SplashBox. Alternatively, you can add one yourself."

We maybe should have never had a list of widgets in the original error hint for the material debug check. If something like that changes in the future, I don't trust myself to remember to check the error hints to be correct.